File: Groove.SP/Groove.SP.Core/Entities/ContractMasterModel.cs
using Groove.SP.Core.Models;
using System;

namespace Groove.SP.Core.Entities
{
    public class ContractMasterModel: Entity
    {
        public long Id { get; set; }
        /// <summary>
        /// To map with <see cref="MasterBillOfLadingModel.CarrierContractNo"/>
        /// </summary>
        public string CarrierContractNo { get; set; }
        /// <summary>
        /// To map with <see cref="Groove.CSFE.Core.Entities.CarrierModel.CarrierCode"/>
        /// </summary>
        public string CarrierCode { get; set; }
        public string RealContractNo { get; set; }
        public string AccountName { get; set; }
        public string ContractType { get; set; }
        public string ColoaderCode { get; set; }
        /// <summary>
        /// It contains two data types:
        /// <list type="number">
        /// <item>If it's matched with <see cref="Groove.CSFE.Core.Entities.OrganizationModel.EdisonCompanyCodeId"/>, store <see cref="Groove.CSFE.Core.Entities.OrganizationModel.Id"/> </item>
        /// <item>If it's not matched, store its own value.</item>
        /// </list>
        /// </summary>
        public string ContractHolder { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime ValidTo { get; set; }
        public ContractMasterStatus Status { get; set; }
        public bool? IsVIP { get; set; }
        public string Remarks { get; set; }
        public string ParentContract { get; set; }
        public string CustomerContractType { get;set; }

    }
}

File: Groove.SP/groove-sp-web/src/app/features/shipment/master-bill-of-lading/models/contract-master-model.ts
import { BaseModel } from 'src/app/core/models/base.model';

export interface ContractMasterModel  extends BaseModel {
    id: number;
    carrierContractNo: string;
    carrierCode: string;
    realContractNo: string;
    accountName: string;
    contractType: string;
    coloaderCode: string;
    contractHolder: string;
    validFrom: string;
    validTo: string;
    status: ContractMasterStatus;
    isVIP: boolean | null;
    remarks: string;
    parentContract: string;
}

export enum ContractMasterStatus {
    Active = 1,
    Inactive = 0
}
